internal/controller/plan/config_template.go

```diff
@@ -147,6 +147,13 @@ func (c *configTemplateBuilder) render(configs map[string]string) (map[string]st
 }
 
 func (c *configTemplateBuilder) builtinObjectsAsValues() (*gotemplate.TplValues, error) {
+	// preHandle the component
+	var v Visitor = &ComponentVisitor{component: c.component}
+	// v = NewDecoratedVisitor(v, resolveServiceReferences(c.cli, c.ctx, c.namespace))
+	if err := v.Visit(resolveServiceReferences(c.cli, c.ctx, c.namespace)); err != nil {
```

Comment:
> Does namespace need to be consistent with cluster? Otherwise, this usage scenario will be restricted. For example, monitoring must be an independent namespace, it may not be possible to access the ServiceRef of the cluster.

fixed in: https://github.com/apecloud/kubeblocks/pull/5112